src/lib/util/permLevels.js

```diff
@@ -33,26 +43,28 @@ class PermissionLevels {
 	 * @returns {PermissionLevels} This permission levels
 	 */
 	addLevel(level, brk, check) {
-		if (this.levels.has(level)) throw new Error(`Level ${level} is already defined`);
-		this.levels.set(level, { break: brk, check: check });
-		return this;
+		return this.set(level, { break: brk, check });
 	}
 
 	/**
-	 * The current valid permissions structure
-	 * @readonly
-	 * @type {validPermStructure}
+	 * Checks if all permission levels are valid
+	 * @return {boolean}
 	 */
-	get structure() {
-		const structure = [];
-		for (let i = 0; i < 11; i++) {
-			const myLevel = this.levels.get(i);
-			if (myLevel) structure.push(myLevel);
-			else structure.push({ break: false, check: () => false });
-		}
-		return structure;
+	isValid() {
+		/* eslint-disable no-multi-spaces, arrow-body-style, operator-linebreak, indent, no-mixed-spaces-and-tabs */
+		return every(this)((level, index) => {
+			// Trick to replace a switch case or if else. Condition on the left, what's executed on the right. Last one is default.
+			return   typeof level !== 'object'   ? throwE(`Permission level ${index} must be an object`) :
+				typeof level.break !== 'boolean'  ? throwE(`"break" in permission level ${index} must be a boolean`) :
+				typeof level.check !== 'function' ? throwE(`"check" in permission level ${index} must be a function`)
+				                                  : true;
+		});
+		/* eslint-enable no-multi-spaces, arrow-body-style, operator-linebreak, indent, no-mixed-spaces-and-tabs */
 	}
 
 }
 
+const throwE = msg => { throw new Error(msg); };
+const every = iterable => func => Array.prototype.every.call(iterable, func);
```

Comment:
Use this instead: https://github.com/hydrabolt/discord.js/blob/stable/src/util/Collection.js#L279-L292